Fix _terms_enum display values

cbuescher opened the pull request:
A recent change (#93839) introduced _terms_enum support for fields that need to convert their internal bytes representation to thr proper string representation for display purposes. The constant_keyword and flattened field types didn't implement a 'valueForDisplay' method yet, so the underlying BytesRef was printed directly in the response.

Closes #94041

cbuescher committed:
Fix _terms_enum display values

A recent change (#93839) introduced _terms_enum support for fields that need to
convert their internal bytes representation to thr proper string representation
for display purposes. The constant_keyword and flattened field types didn't
implement a 'valueForDisplay' method yet, so the underlying BytesRef was printed
directly in the response.

Closes #94041

elasticsearchmachine commented:
Pinging @elastic/es-search (Team:Search)

elasticsearchmachine commented:
Hi @cbuescher, I've created a changelog YAML for you.

cbuescher committed:
Update docs/changelog/94080.yaml

romseygeek reviewed (approved):
LGTM.  It might be worth, in a follow-up, creating a test in MapperTestCase that everything that supports terms can enable, so that we ensure we keep coverage on this?

cbuescher commented:
> It might be worth, in a follow-up, creating a test in MapperTestCase that everything that supports terms can enable, so that we ensure we keep coverage on this?

The problem here is that the conversion to a human-readable representation of the BytesRef underlying the terms_enum functionality happens in the transport action, not the mappers, so any testing about that would have to happen higher up in the call chain. Thats why I added some simple rest tests here, I was surprised about this gap in testing the return values for different field types as well and found that the existing "10_basic" yaml test in for terms_enum is a bit to complex to really add simple, easy to understand tests, so I added the one in this PR.
